### DISCUSSION TITLE
commit patterns

### DISCUSSION THREAD

**Discussion by kauaneiras** [community/community#74403] [Other Feature Feedback, Questions, & Ideas]
### What are the best commit patterns to keep my codes organized?

**Comment by EliasAfara** [comment]
I personally use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) which is a standardized format for writing commit messages in version control systems like Git. It aims to provide a consistent and structured way of documenting changes to a codebase, making it easier to automate release notes generation, track issues, and understand the history of a project.

## Conventional Commit Format

The format of a Conventional Commit message typically consists of three parts:

1. **Type:** This is a mandatory field that describes the purpose or nature of the commit. Common types include:

   - `feat`: Signifies the addition of a new feature.
   - `fix`: Indicates the correction of a bug or issue.
   - `refactor`: Denotes changes to the codebase that do not add new features or fix issues but improve the code's structure or readability.
   - `docs`: Refers to updates or additions to documentation.
   - `style`: Represents changes related to code styling, such as formatting or styling rules.
   - `test`: Indicates the addition or modification of tests.
   - `chore`: Covers tasks like dependency updates, build configuration changes, or general maintenance.

2. **Scope (optional):** This field provides additional context, specifying which part of the codebase the commit affects. It's particularly useful in larger projects with multiple components or modules. For example, you might specify the scope as a component name or a specific file.

3. **Message:** The message should be a brief, concise, and meaningful description of the change or action taken. It explains what the commit does and, if necessary, why the change is made.

Additionally, Conventional Commits also allow for a few other special commit types that have specific purposes:

- **Breaking Changes (breaking):** Used to indicate that the commit introduces breaking changes. It's essential to provide detailed information about what's changed and how it affects existing code.

- **Revert (revert):** Indicates that the commit reverts a previous commit. The message should explain which commit is being reverted and why.

Conventional Commits are designed to make it easier to generate changelogs, automate version numbering, and provide a clear and structured history of changes in a codebase. Various tools and libraries, such as standard-version and semantic-release, can automate these processes by analyzing commit messages following the Conventional Commits format.
